zapcore/encoder.go

```diff
@@ -331,6 +332,9 @@ type EncoderConfig struct {
 	// Unlike the other primitive type encoders, EncodeName is optional. The
 	// zero value falls back to FullNameEncoder.
 	EncodeName NameEncoder `json:"nameEncoder" yaml:"nameEncoder"`
+	// Configure the encoder for interface{} type objects
+	// If not provided, objects are encoded using json.Encoder
+	NewReflectedEncoder func(io.Writer) ReflectedEncoder `json:"newReflectedEncoder" yaml:"newReflectedEncoder"`
```

Comment:
Since this a function that is not JSON/YAML encodeable, we'll have to use `-` here.
```suggestion
	NewReflectedEncoder func(io.Writer) ReflectedEncoder `json:"-" yaml:"-"`
```